### DISCUSSION TITLE
Memory management for Objective-C code via cgo

### DISCUSSION THREAD

**Discussion by tejaskumark** [progrium/darwinkit#194] [Q&A]
Hi Experts, 

I am kind of starting on Mac OS and Golang intersection. I was going through the memory management article this repo has and it gives good starting point on memory management when working with Go and Mac OS Objective C API. But I have question like,
 - When we compile go code with Objective C API, which memory management will be used MMR or ARC?
 - How golang compiler decides ? If there are any flags to pass while compiling with `go build` that instructs to use MMR or ARC?

**Comment by progrium** [comment]
ARC is a feature of the Objective-C compiler, which does not compile Go code. Therefore, the only option is MMR. Please let me know how to make this more clear in the documentation.

**Reply by tejaskumark** [reply]
@progrium I am not sure if this flag denotes ARC or not , but I was trying to compile with go build with `// #cgo CFLAGS: -x objective-c -fobjc-arc` and it was compiling with and without this flag as expected. I was trying to get help on memory management part on Apple Dev forum(https://developer.apple.com/forums/thread/735926) for Go lang compilation, but I found your write up giving some info, but if you can add bit more practical info with explaining go build flags and all would be good. I understand this info should be on golang site but for now I am not able to understand memory management part on objective-C. Will again read memory management article of this repo and check if my understanding improves.

**Reply by progrium** [reply]
I see now you are talking about Objective-C code using cgo. I'm not sure if there is a flag you can pass via cgo that will make it compile with ARC. I am currently under the assumption (which I have not explicitly validated, but haven't seen evidence otherwise) that, at least by default, Objective-C code compiled via cgo will use MMR. Since this is the model used when writing Go code, I don't think I would encourage using ARC for cgo Objective-C code anyway. However, from this project's perspective this is an advanced technique outside the scope of our help. That said, let us know if you find something out.

**Reply by tejaskumark** [reply]
@progrium Thanks for the explanation. Will update if I find out something more on this.
